blueprints/third-party-solutions/phpipam/variables.tf

```diff
@@ -0,0 +1,154 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# Documentation: https://cloud.google.com/run/docs/securing/managing-access#making_a_service_public
+variable "cloud_run_invoker" {
+  type        = string
+  description = "IAM member authorized to access the end-point (for example, 'user:YOUR_IAM_USER' for only you or 'allUsers' for everyone)."
+  default     = "allUsers"
+}
+
+variable "cloudsql_password" {
+  type        = string
+  description = "CloudSQL password (will be randomly generated by default)."
+  default     = null
+}
+
+variable "connector" {
+  type        = string
+  description = "Existing VPC serverless connector to use if not creating a new one."
+  default     = null
+}
+
+variable "create_connector" {
+  type        = bool
+  description = "Should a VPC serverless connector be created or not."
+  default     = true
+}
+
+variable "custom_domain" {
+  type        = string
+  description = "Cloud Run service custom domain for GLB."
+  default     = null
+}
+
+variable "iap" {
+  description = "Identity-Aware Proxy for Cloud Run in the LB."
+  type = object({
+    enabled            = optional(bool, false)
+    app_title          = optional(string, "Cloud Run Explore Application")
+    oauth2_client_name = optional(string, "Test Client")
+    email              = optional(string)
+  })
+  default = {}
+}
+
+# PSA: documentation: https://cloud.google.com/vpc/docs/configure-private-services-access#allocating-range
+variable "ip_ranges" {
+  description = "CIDR blocks: VPC serverless connector, Private Service Access(PSA) for CloudSQL, CloudSQL VPC."
+  type = object({
+    connector = string
+    psa       = string
+    ilb       = string
+  })
+  default = {
+    connector = "10.8.0.0/28"
+    psa       = "10.60.0.0/24"
+    ilb       = "10.128.0.0/28"
+  }
+}
+
+variable "phpipam_config" {
+  type = object({
+    image = optional(string, "phpipam/phpipam-www:latest")
+    port  = optional(number, 80)
+  })
+  description = "PHPIpam configuration."
+  default = {
+    image = "phpipam/phpipam-www:latest"
+    port  = 80
+  }
+}
+
+variable "phpipam_exposure" {
+  type        = string
+  description = "Whether to expose the application publicly via GLB or internally via ILB, default GLB."
+  default     = "EXTERNAL"
+  validation {
+    condition     = var.phpipam_exposure == "INTERNAL" || var.phpipam_exposure == "EXTERNAL"
+    error_message = "phpipam_exposure supports only 'INTERNAL' or 'EXTERNAL'"
+  }
+}
+
+variable "phpipam_password" {
+  type        = string
+  description = "Password for the phpipam user (will be randomly generated by default)."
+  default     = null
+}
+
+variable "prefix" {
+  description = "Prefix used for resource names."
+  type        = string
+  validation {
+    condition     = var.prefix != ""
+    error_message = "Prefix cannot be empty."
+  }
+}
+
+variable "principals" {
```

Comment:
principals is too vague, can we call this something like `admin_identities`? Also, what if I want to use a group and/or service account?

Reply:
Tried adding admin_identities as an object with users, service accounts and groups as list and tested with different identity kinds